### PR TITLE
codec(ticdc): avro encode watermark event to be compatible with consumer

### LINKED ISSUE #8869: add watermark event to the avro encoder to support kafka consumer sink

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #8869

### What is changed and how it works?

Originally, avro does not support encode DDL and checkpoint event, which means the downstream kafka consumer does not the DDL exsitance and data stream progress, this is incompatible with the current cdc kafka consumer.

* add a new config in the sink url, named `avro-enable-watermark`
* encoding DDL and checkpoint event so that can be sink to the kafka

#### Tests 
- Unit test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Comment by asddongmen** [comment]
Could you add some descriptions in the issue this PR is referred to? To help others understand the purpose of this PR. Thank you~

**Review by asddongmen — APPROVED** [review]

File: pkg/sink/codec/common/config_test.go [hunk 1]
```diff
@@ -88,6 +88,25 @@ func TestConfigApplyValidate4EnableRowChecksum(t *testing.T) {
...
+	uri = "kafka://127.0.0.1:9092/avro?protocol=avro&enable-tidb-extension=true&avro-enable-watermark=true"
+	sinkURI, err = url.Parse(uri)
+	require.NoError(t, err)
+
+	protocol = sinkURI.Query().Get("protocol")
+	p, err = config.ParseSinkProtocolFromString(protocol)
+	require.NoError(t, err)
+	c = NewConfig(p)
+
+	replicaConfig = config.GetDefaultReplicaConfig()
```
Comment (nongfushanquan) [inline pkg/sink/codec/common/config_test.go:102]: could you add some cases to verify the watermark and ddl event

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**